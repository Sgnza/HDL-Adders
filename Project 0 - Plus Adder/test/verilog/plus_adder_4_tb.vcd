$date
	Tue Jan 16 15:15:54 2024
$end
$version
	Icarus Verilog
$end
$timescale
	1ps
$end
$scope module plus_adder_4_tb $end
$var wire 4 ! sum_tb [3:0] $end
$var wire 1 " cout_tb $end
$var reg 4 # a_tb [3:0] $end
$var reg 4 $ b_tb [3:0] $end
$var reg 1 % cin_tb $end
$scope module pa0 $end
$var wire 4 & a [3:0] $end
$var wire 4 ' b [3:0] $end
$var wire 1 % cin $end
$var wire 5 ( tmp [4:0] $end
$var wire 4 ) sum [3:0] $end
$var wire 1 " cout $end
$upscope $end
$scope task verify $end
$var reg 4 * a_task [3:0] $end
$var reg 4 + b_task [3:0] $end
$var reg 1 , cin_task $end
$var reg 1 - cout_task $end
$var reg 4 . sum_task [3:0] $end
$upscope $end
$upscope $end
$enddefinitions $end
$comment Show the parameter values. $end
$dumpall
$end
#0
$dumpvars
b0 .
1-
0,
b1 +
b1111 *
b0 )
b10000 (
b1 '
b1111 &
0%
b1 $
b1111 #
1"
b0 !
$end
#1
