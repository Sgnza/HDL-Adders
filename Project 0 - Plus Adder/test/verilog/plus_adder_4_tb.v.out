#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "C:\iverilog\lib\ivl\system.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "C:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "C:\iverilog\lib\ivl\va_math.vpi";
S_0000017e9b935e60 .scope module, "plus_adder_4_tb" "plus_adder_4_tb" 2 3;
 .timescale -12 -12;
v0000017e9b997140_0 .var "a_tb", 3 0;
v0000017e9b9976e0_0 .var "b_tb", 3 0;
v0000017e9b997780_0 .var "cin_tb", 0 0;
v0000017e9b9973c0_0 .net "cout_tb", 0 0, L_0000017e9b996b00;  1 drivers
v0000017e9b996c40_0 .net "sum_tb", 3 0, L_0000017e9b996a60;  1 drivers
S_0000017e9b9359b0 .scope module, "pa0" "plus_adder_4" 2 11, 3 1 0, S_0000017e9b935e60;
 .timescale -12 -12;
    .port_info 0 /INPUT 4 "a";
    .port_info 1 /INPUT 4 "b";
    .port_info 2 /INPUT 1 "cin";
    .port_info 3 /OUTPUT 4 "sum";
    .port_info 4 /OUTPUT 1 "cout";
v0000017e9b912d00_0 .net *"_ivl_0", 4 0, L_0000017e9b996ec0;  1 drivers
v0000017e9b93a950_0 .net *"_ivl_10", 4 0, L_0000017e9b996920;  1 drivers
L_0000017e9b99f8e8 .functor BUFT 1, C4<0000>, C4<0>, C4<0>, C4<0>;
v0000017e9b93a9f0_0 .net *"_ivl_13", 3 0, L_0000017e9b99f8e8;  1 drivers
L_0000017e9b99f858 .functor BUFT 1, C4<0>, C4<0>, C4<0>, C4<0>;
v0000017e9b93aa90_0 .net *"_ivl_3", 0 0, L_0000017e9b99f858;  1 drivers
v0000017e9b93ae30_0 .net *"_ivl_4", 4 0, L_0000017e9b997320;  1 drivers
L_0000017e9b99f8a0 .functor BUFT 1, C4<0>, C4<0>, C4<0>, C4<0>;
v0000017e9b93aed0_0 .net *"_ivl_7", 0 0, L_0000017e9b99f8a0;  1 drivers
v0000017e9b93af70_0 .net *"_ivl_8", 4 0, L_0000017e9b996880;  1 drivers
v0000017e9b93b010_0 .net "a", 3 0, v0000017e9b997140_0;  1 drivers
v0000017e9b93b0b0_0 .net "b", 3 0, v0000017e9b9976e0_0;  1 drivers
v0000017e9b997460_0 .net "cin", 0 0, v0000017e9b997780_0;  1 drivers
v0000017e9b9975a0_0 .net "cout", 0 0, L_0000017e9b996b00;  alias, 1 drivers
v0000017e9b996ba0_0 .net "sum", 3 0, L_0000017e9b996a60;  alias, 1 drivers
v0000017e9b997280_0 .net "tmp", 4 0, L_0000017e9b996f60;  1 drivers
L_0000017e9b996ec0 .concat [ 4 1 0 0], v0000017e9b997140_0, L_0000017e9b99f858;
L_0000017e9b997320 .concat [ 4 1 0 0], v0000017e9b9976e0_0, L_0000017e9b99f8a0;
L_0000017e9b996880 .arith/sum 5, L_0000017e9b996ec0, L_0000017e9b997320;
L_0000017e9b996920 .concat [ 1 4 0 0], v0000017e9b997780_0, L_0000017e9b99f8e8;
L_0000017e9b996f60 .arith/sum 5, L_0000017e9b996880, L_0000017e9b996920;
L_0000017e9b996a60 .part L_0000017e9b996f60, 0, 4;
L_0000017e9b996b00 .part L_0000017e9b996f60, 4, 1;
S_0000017e9b936310 .scope task, "verify" "verify" 2 19, 2 19 0, S_0000017e9b935e60;
 .timescale -12 -12;
v0000017e9b996e20_0 .var "a_task", 3 0;
v0000017e9b9970a0_0 .var "b_task", 3 0;
v0000017e9b9969c0_0 .var "cin_task", 0 0;
v0000017e9b9971e0_0 .var "cout_task", 0 0;
v0000017e9b997640_0 .var "sum_task", 3 0;
TD_plus_adder_4_tb.verify ;
    %load/vec4 v0000017e9b996e20_0;
    %store/vec4 v0000017e9b997140_0, 0, 4;
    %load/vec4 v0000017e9b9970a0_0;
    %store/vec4 v0000017e9b9976e0_0, 0, 4;
    %load/vec4 v0000017e9b9969c0_0;
    %store/vec4 v0000017e9b997780_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000017e9b997640_0;
    %load/vec4 v0000017e9b996c40_0;
    %cmp/e;
    %flag_get/vec4 4;
    %jmp/0 T_0.2, 4;
    %load/vec4 v0000017e9b9971e0_0;
    %load/vec4 v0000017e9b9973c0_0;
    %cmp/e;
    %flag_get/vec4 4;
    %and;
T_0.2;
    %flag_set/vec4 8;
    %jmp/0xz  T_0.0, 8;
    %vpi_call 2 32 "$display", "CORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de entrada cin: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", v0000017e9b997140_0, v0000017e9b9976e0_0, v0000017e9b9969c0_0, v0000017e9b997640_0, v0000017e9b996c40_0, v0000017e9b9971e0_0, v0000017e9b9973c0_0 {0 0 0};
    %jmp T_0.1;
T_0.0 ;
    %vpi_call 2 34 "$display", "INCORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de entrada cin: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", v0000017e9b997140_0, v0000017e9b9976e0_0, v0000017e9b9969c0_0, v0000017e9b997640_0, v0000017e9b996c40_0, v0000017e9b9971e0_0, v0000017e9b9973c0_0 {0 0 0};
T_0.1 ;
    %end;
    .scope S_0000017e9b935e60;
T_1 ;
    %vpi_call 2 42 "$dumpfile", "plus_adder_4_tb.vcd" {0 0 0};
    %vpi_call 2 43 "$dumpvars", 32'sb00000000000000000000000000000000, S_0000017e9b935e60 {0 0 0};
    %end;
    .thread T_1;
    .scope S_0000017e9b935e60;
T_2 ;
    %pushi/vec4 15, 0, 4;
    %store/vec4 v0000017e9b996e20_0, 0, 4;
    %pushi/vec4 1, 0, 4;
    %store/vec4 v0000017e9b9970a0_0, 0, 4;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000017e9b9969c0_0, 0, 1;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0000017e9b997640_0, 0, 4;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000017e9b9971e0_0, 0, 1;
    %fork TD_plus_adder_4_tb.verify, S_0000017e9b936310;
    %join;
    %end;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "plus_adder_4_tb.v";
    "./../../src/verilog/plus_adder_4.v";
